### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by Feliverse**
**Project requirements**

- Connect your store to the main component of your React app.
- Define a state slice which will contain your books

```
// Initial state:
[
  {
    "item_id": "item1",
    "title": "The Great Gatsby",
    "author": "John Smith",
    "category": "Fiction"
  },
  {
    "item_id": "item2",
    "title": "Anna Karenina",
    "author": "Leo Tolstoy",
    "category": "Fiction"
  },
  {
    "item_id": "item3",
    "title": "The Selfish Gene",
    "author": "Richard Dawkins",
    "category": "Nonfiction"
  }
]
```

- Display your `books`, received from the slice, in a reusable component
- Dispatch actions using `useDispatch`
- Add a `<Button>` component, which includes:
> An event handler that adds a book to the `books` array (with attributes id, title and author)
- Add a `<Button>` component, which includes: 
> An event handler that removes a book from the `books` array (by id)
- Add a `<Button>` component, which includes:
> An event handler that check status (i.e add a functionality for "Check status" button click)

**Need a big picture?**
Remind me about [the big picture of this project](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/sneak_peek_v2_0.md).

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
         <li className="option">Comments</li>
-        <li className="option">Remove</li>
+        <li>
+          <button
+            className="option"
+            onClick={() => { handleDelete(id); }}
+            onKeyDown={() => { handleDelete(id); }}
+            type="button"
+          >
+            Remove
```
Comment (omar-muhamad) [inline src/components/Book.js]: - [ ] When I try to remove a book, all books get removed, that's not supposed to happen, Please fix that.

chrome-capture-2023-2-9

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Feliverse,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Book.js (hunk 1 above)
Comment (Feliverse) [reply src/components/Book.js]: Thank you, I'll fix it

File: src/components/Form.js [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
+  const handlesubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook({ next: book }));
+    setbook({ title: '', item_id: '', author: '' });
+  };
+
   return (
     <div>
-      <form className="newBook">
+      <form className="addNew" onSubmit={handlesubmit}>
```
Comment (FrederickMih) [inline src/components/Form.js:18]: - [Optional] This event handler was supposed to be implemented on the button below(onClick) and not on the form itself. You can check the requirement once more.

File: src/components/Book.js [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
         <li className="option">Comments</li>
-        <li className="option">Remove</li>
+        <li>
+          <button
+            className="option"
+            onClick={() => { handleDelete(id); }}
+            type="button"
+          >
+            Remove
+          </button>
```
Comment (FrederickMih) [inline src/components/Book.js:26]: :+1:

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Feliverse, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- PR is professional :+1: 
- Add and remove book functionalities still work as expected :+1: 
- Linter checks passed :heavy_check_mark: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**